src/store/global/global.thunks.ts

```diff
@@ -440,6 +441,68 @@ function checkPendingDelayedWithdrawals(): AppThunk {
   };
 }
 
+/**
+ * Checks if we have some exit that is not saved in localStorage
+ */
+function recoverPendingDelayedWithdrawals(exits: Exits): AppThunk {
+  return (dispatch: AppDispatch, getState: () => AppState) => {
+    const {
+      global: { wallet, pendingDelayedWithdraws, ethereumNetworkTask },
+    } = getState();
+    if (wallet !== undefined && ethereumNetworkTask.status === "successful") {
+      const {
+        data: { chainId },
+      } = ethereumNetworkTask;
+      if (chainId !== undefined) {
+        const { hermezEthereumAddress } = wallet;
+        const provider = Providers.getProvider();
+        const {
+          data: { chainId },
+        } = ethereumNetworkTask;
+        const accountPendingDelayedWithdraws: PendingDelayedWithdraw[] =
+          storage.getPendingDelayedWithdrawsByHermezAddress(
+            pendingDelayedWithdraws,
+            chainId,
+            hermezEthereumAddress
+          );
+        const batchNumPendingDelayedWithdraws = accountPendingDelayedWithdraws.map(
+          (account) => account.batchNum
+        );
+
+        exits.exits.map(async (exit) => {
```

Comment:
It feels strange using `map` here since no transformation is required. I mean, the result of this map is a list of promises that we don't return. I think that using `forEach` feels more correct.

Reply:
I know but I prefer use map instead of for of because we can't use async inside forEach